app/airview_api/controllers/search.py

```diff
@@ -0,0 +1,31 @@
+from flask import request
+from flask.views import MethodView
+
+from airview_api.blueprint import Blueprint, Roles
+from airview_api.schemas import SearchResultSchema
+from airview_api.search import init_search_backend
+
+blp = Blueprint(
+    "search",
+    __name__,
+    url_prefix="/search",
+    description="Search endpoint to query existing documents."
+)
+
+
+search_backend = init_search_backend()
+
+
+@blp.route("/")
+class Search(MethodView):
+    @blp.response(200, SearchResultSchema(many=True))
+    @blp.role(Roles.CONTENT_READER)
+    def get(self):
+        """
+        Searches matching terms in a configured search backend.
+        """
+        results = []
+        q = request.args.get("q", None)
```

Comment:
Needs opt params to handle search and result behaviour (i.e.: number of returned results, wildcards..)

Reply:
Have a look to see if the flask-smorest schema stuff will help, there's an exaple - https://github.com/AirWalk-Digital/airview-api/blob/main/app/airview_api/controllers/technical_controls.py#L34. It allows for deserialization of query params into the schema and handles all the param validation, etc.